queue/sqs.go

```diff
@@ -174,6 +176,10 @@ func (q *sqsFIFOQueue) Next(ctx context.Context) amboy.Job {
 	if err != nil {
 		return nil
 	}
+	if err := q.scopes.Acquire(job.ID(), job.Scopes()); err != nil {
+		q.Put(ctx, job)
```

Comment:
Why call q.Put when acquisition fails?

Reply:
because otherwise failing to get the lock (ususally because someone else has it) will cause us to never do this job.